fenicsprecice/adapter_core.py

```diff
@@ -415,31 +412,36 @@ def get_coupling_triangles(function_space, coupling_subdomain, precice_edge_dict
         Function space on which the finite element problem definition lives.
     coupling_subdomain : FEniCS Domain
         FEniCS domain of the coupling interface region.
-    precice_edge_dict: dict
-        Dictionary with FEniCS IDs of coupling mesh edges as keys and preCICE IDs of the edges as values
+    fenics_edge_ids: numpy array
+        Array with FEniCS IDs of coupling mesh edges
 
     Returns
     -------
-    edges : numpy array
-        Array of edges indices (3 per triangle)
+    vertex_ids : numpy array
+        Array of indices of vertices which make up triangles (3 per triangle)
     """
 
     def cell_is_in(subdomain, this_cell):
         """
         Check whether edge lies within subdomain
         """
-        assert(len(list(vertices(this_cell))) == 3), "Only triangular meshes are supported"
+        assert (len(list(vertices(this_cell))) == 3), "Only triangular meshes are supported"
         return all([subdomain.inside(v.point(), True) for v in vertices(this_cell)])
 
-    edges_ids = []
-
+    vertex_ids = []
     for cell in cells(function_space.mesh()):
         if cell_is_in(coupling_subdomain, cell):
             e1, e2, e3 = list(edges(cell))
-            if all(edge in precice_edge_dict.keys() for edge in [e1.index(), e2.index(), e3.index()]):
-                edges_ids.append([e1.index(), e2.index(), e3.index()])
-
-    return np.array(edges_ids)
+            if all(edge_ids in fenics_edge_ids for edge_ids in [e1.index(), e2.index(), e3.index()]):
+                v1, v2 = vertices(e1)
+                _, v3 = vertices(e2)
+                assert (v3 != v1)
+                assert (v2 != v2)
```

Comment:
I found this problem by running `channel-transport-reaction`. But even fixing this obviously wrong assertion does not help. There seems to be some issue with the function `get_coupling_triangles`. We obviously need a test for this function and need to fix it. @IshaanDesai can you take a look?